proto/evm/v1beta1/tx.proto

```diff
@@ -20,6 +20,47 @@ message SetGatewayRequest {
 
 message SetGatewayResponse {}
 
+message ConfirmGatewayTxRequest {
+  bytes sender = 1 [ (gogoproto.casttype) =
+                         "github.com/cosmos/cosmos-sdk/types.AccAddress" ];
+  string chain = 2;
+  bytes tx_id = 3 [
+    (gogoproto.nullable) = false,
+    (gogoproto.customtype) = "Hash",
+    (gogoproto.customname) = "TxID"
+  ];
+}
+
+message ConfirmGatewayTxResponse {}
+
+message VoteConfirmGatewayTxRequest {
+  message Vote {
+    message Event {
+      evm.v1beta1.EventInfo info = 1 [ (gogoproto.nullable) = false ];
+      oneof event {
+        evm.v1beta1.EventContractCallWithToken contract_call_with_token = 2;
+      }
```

Comment:
This seems a bit overly complicated (and as mentioned earlier, the nesting leads to terrible naming). We should probably discuss if this is necessary